src/utils.ts

```diff
@@ -389,6 +393,33 @@ export const trimUrl = (url: string): string => {
     .join('/')
 }
 
+/**
+ * Takes a `ethers.Wallet` instance and returns a object of internal `SigningKey`.
+ * @param walletFromEthers `ethers.Wallet` instance.
+ */
+export function getSigningKeyFromEthers(
+  walletFromEthers: ethers.Wallet
+): SigningKey {
+  return {
+    mnemonic: walletFromEthers.mnemonic,
+    privateKey: walletFromEthers.privateKey,
+    derivationPath: walletFromEthers.path
+  }
+}
+
+/**
+ * Takes a `TLWallet` of type `ethers` or `identity` and returns an instance of `ethers.Wallet`.
+ * @param wallet `TLWallet` of type `ethers` or `identity`.
+ */
+export function getWalletFromEthers(
+  wallet: EthersWalletSchema | IdentityWalletSchema
+): ethers.Wallet {
+  const { signingKey } = wallet.meta
+  return signingKey.mnemonic && signingKey.derivationPath
+    ? ethers.Wallet.fromMnemonic(signingKey.mnemonic, signingKey.derivationPath)
```

Comment:
Isnt that too slow? isnt the private key in that case recalculated from the mnemonic? We should always recover from the private key directly and additionally restore the mnemonic.